### ISSUE TITLE
Grotto before non-floor

### ISSUE THREAD

**Issue by kensanata** [kensanata/gridmapper#35]
When a non-floor is added *after* a grotto tile, the grotto tile does not adapt. Simplest example: compare `t g` with `gt`. This is a problem when loading files.

**Comment by kensanata** [comment]
You can see this in the [OLIVING](https://campaignwiki.org/gridmapper.svg?load=OLIVING) map.

```
(28,3)f

        wwww fG wwww wwww wwww wwww wwww wwww wwww wwww wwww wwww wwww
       www gf gbvB pvvvv g gcv g c w
       www  gggpvvvv pvvvv pvvvv pvvvv  g4fgdvgw
       www g g pvvvv t t pvvvv g gg w
       www f3gdvg pvvvv t t pvvvv pvvvv g ggw
       www g g pvvvv t t t pvvvv fwvvvvg5fgw
       www  gggpvvvv t t t pvvvv g g w
       www  g  gpvvvv t pvvvv 2g gg w
       www  ggg gpvvvv gg g6fgw
       www  ddddvg gg   w.w ww ww ww ww
       www gfg g ggw
       www cccvvv 1ff ggffw
       www pv ffg  ggfG
       www  gcv c g   w
        ww ww ww ww ww ww ww ww(28,11,0)
```

**Comment by kensanata** [comment]
![screen shot 2017-05-13 at 21 33 03](https://cloud.githubusercontent.com/assets/1145712/26028563/bc9d4ddc-3823-11e7-8cee-7b32f9ed2eef.png)

**Comment by kensanata** [comment]
Fixed:
![screen shot 2017-05-13 at 21 33 40](https://cloud.githubusercontent.com/assets/1145712/26028567/d13cbf66-3823-11e7-8e85-5afc468a9417.png)

**Comment by kensanata** [comment]
Fixed by 510cfdd.
